Add BLAS/LAPACK to Fedora installation. Fixes #159

fasiha opened the pull request:
Advises users to Fedora/CentOS/etc. to install dev versions of BLAS/LAPACK so that subsequent upgrades via `pip` will not fail.

charris commented:
Hmm, we are now putting linux binaries up on PiPI, so this may not be required. I'd also recommend atlas-devel instead of blas, and dnf instead of yum for recent fedora. Not sure where all that should go. @rgommers Thoughts?

rgommers commented:
Not a Fedora user, so no opinion on ``dnf`` vs ``yum. 

``atlas-devel`` (assuming it works) does sound preferable. @fasiha can you check if that works for you, and if so change to that?

takluyver commented:
`dnf` is the new `yum` on recent versions of Fedora (Fedora >= 23, I think). `yum` is aliased to it, so yum commands will still work for now. I'd guess it will remain for a decent length of time.

charris commented:
@rgommers Fedora moved to `dnf` in Fedora 22, so it is fairly new.

fasiha commented:
@rgommers sorry took a few days. `atlas-devel` and `lapack-devel` tested to work (atlas-devel has blas-devel as prerequisite). I tested on my ancient CentOS 6.5 VM, so I have no preference on `dnf` vs `yum`. Updated the branch and rebased to grab recent changes.

rgommers commented:
If ``dnf`` is new and ``yum`` still works, let's stay with ``yum`` for a little while I'd say - it works for a wider range of OS versions.

rgommers commented:
> Now that I look this big list of changes my PR is proposing… should I not have rebased my branch on master? Sorry for being open-source noob

No worries. Something did go wrong here, not sure how you managed to get those commits in here. You can either just add commits to an existing PR, or rebase on master and then force-push to the open PR. But it looks like you cherry-picked from master?

rgommers commented:
@fasiha this should work without ``lapack-devel``, ATLAS provides an optimized LAPACK. Can you confirm that that works for you?

rgommers commented:
Don't really want to spend the time fixing up this PR, since it's a one-liner. I cherry-picked the relevant commits, squashed them, removed ``lapack-devel`` and pushed the commit in f53c1ff01f.
